src/vault/client/mock.clj

```diff
@@ -59,21 +59,41 @@
 
   (create-token!
     [this opts]
-    {:request-id ""
-     :lease-id ""
-     :renewable false
-     :lease-duration 0
-     :data nil
-     :wrap-info
-     (when (:wrap-ttl opts)
-       (let [wrap-token (gen-uuid)]
-         (swap! cubbies assoc wrap-token (mock-token-auth))
-         {:token wrap-token
-          :ttl (:wrap-ttl opts)
-          :creation-time (gen-date)
-          :wrapped-accessor (gen-uuid)}))
-     :warnings nil
-     :auth (when-not (:wrap-ttl opts) (mock-token-auth))})
+    (let [dates {\s 1 \m 60 \h 3600 \d 86400}]
+      (if (:wrap-ttl opts)
+        (let [wrap-token (gen-uuid)]
+          (swap! cubbies assoc wrap-token (mock-token-auth))
+          {:token            wrap-token
+           :ttl              (* (read-string (subs (:wrap-ttl opts) 0 (dec (count (:wrap-ttl opts)))))
+                                (get dates (last (:wrap-ttl opts))))
```

Comment:
What if there's no time unit? Guarding with `1` as the not-found lookup would be safer.

Reply:
I couldn't tell from the docs if that was a valid call, and we only want to allow valid calls through the mock client so that it fails when the http client would fail. Do you know if it's valid? If not, I'll do some more digging and find out

https://www.vaultproject.io/api/auth/token/index.html#ttl

Reply:
Yeah that makes sense to not be too permissive. I guess I'm thinking of avoiding an (opaque) `NullPointerException` but otherwise throwing if its missing the suffix makes sense to me.

Reply:
True, I'll throw an error with more info :)